### PR TITLE
DevEx: Split tests and combine their coverage; fix SonarCloud coverage

### PR DESCRIPTION
**Pull request by sutt0n**
### Brief
- split tests using concurrent builds _and_ parallelism for both UI unit and integration tests
- combined persisted coverage (via CircleCI workspaces) with `npx istanbul-combine`
- since SonarCloud / SonarQube wasn't reading jest's lcov.info reports correctly, these reports were normalized via `istanbul-combine` as well (and appeared to work -- coverage went from 0.0% to the normal percentage in all 3 projects: shared, api, and client)

### PR CI requirements
- removed `build-client`
- added `build-client-integration`, `build-client-unit`, and `build-client-coverage` jobs

image

### CODE REVIEW

File: web-client/jest-integration.config.js [hunk 1]
```diff
@@ -4,19 +4,32 @@ module.exports = {
...
+    '!src/index-public.prod.js',
   ],
   coverageDirectory: './coverage-integration',
   coverageThreshold: {
     global: {
-      branches: 50,
-      functions: 50,
-      lines: 50,
-      statements: 50,
+      branches: 40,
```
Comment (rachaelparris) [inline web-client/jest-integration.config.js:22]: Why are these numbers being reduced to 40? Do we even need to check coverage at this level?

Comment (sutt0n) [reply web-client/jest-integration.config.js:22]: They were failing because of the split; furthermore, no, we probably don't at this point.

Comment (rachaelparris) [reply web-client/jest-integration.config.js:22]: I think it would probably be best to just remove these if we can if they're pretty much arbitrary.

**Review by rachaelparris — APPROVED** [review]
Looks good to me, but it seems like Github is requiring build-client to pass to merge it?

**Review by matthopson — APPROVED** [review]

**Review by ericsorenson — APPROVED** [review]
🎉

**Pull request merged**